File: Sources/BraintreeCard/BTCardRequest.m
#if __has_include(<Braintree/BraintreeCard.h>)
#import <Braintree/BTCardRequest.h>
#else
#import <BraintreeCard/BTCardRequest.h>
#endif

@implementation BTCardRequest

- (instancetype)initWithCard:(BTCard *)card {
    if (!card) {
        return nil;
    }
    if (self = [super init]) {
        _card = card;
    }
    return self;
}

@end

File: Sources/BraintreeCard/BTCardRequest.swift
import Foundation

/// Contains information about a card to tokenize
// TODO: - NEXT_MAJOR_VERSION remove this class
@available(*, deprecated, message: "Use BTCard directly instead")
@objcMembers public class BTCardRequest: NSObject {

    /// The `BTCard` associated with this instance.
    public var card: BTCard

    /// Initialize a Card request with a `BTCard`.
    /// - Parameter card: The `BTCard` to initialize with.
    public init(card: BTCard) {
        self.card = card
    }
}
